sycl/source/detail/device_info.hpp

```diff
@@ -345,7 +345,8 @@ struct get_device_info_impl<std::vector<kernel_id>,
     std::vector<kernel_id> ids;
     ids.reserve(names.size());
     for (const auto &name : names) {
-      ids.push_back(ProgramManager::getInstance().getBuiltInKernelID(name));
+      ids.push_back(
+          ProgramManager::getInstance().getBuiltInKernelID(name.c_str()));
```

Comment:
We need to create a string in this part of the code anyway, so we can keep passing a `const std::string&` here.

Reply:
Do you mean we should revert getBuiltInKernelID() to take  const std::string&?

Reply:
Seems so, no point to work with char * in this case.